fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,180 @@
+                     capsize=10,
+                     colormap="Pastel1",
+                     subplot_shape=None,
+                     figsize=None
+                     ):
+    """Visualization for metrics with statistical error bounds.
+
+    Plots a given metric and its given error (as described by the `error_mapping`)
+
+    This function takes in a :class:`fairlearn.metrics.MetricFrame` with precomputed metrics and metric errors
+    and a `error_mapping` dict to interpret the columns (??) of the `MetricFrame`.
+    
+    Parameters
+    ----------
+    metric_frame : fairlearn.metrics.MetricFrame
+        The collection fo disaggregated metric values, along with the metric errors.
+
+    plot_type : str
+        The type of plot to display. i.e. "bar", "line", etc
+
+    metrics : str or list of str
+        The name of the metrics to plot. Should match columns from the given `MetricFrame`
+
+    axs : matplotlib.axes._axes.Axes, optional
+        Custom Matplotlib axes to which this function can plot
+
+    show_plot : bool, optional
+        Whether to show the plot or not
+
+    plot_error_labels : bool, optional
+        Whether or not to plot numerical labels for the error bounds
+
+    text_precision_digits : int, optional
+        The number of digits of precision to show for error labels
+
+    text_font_size : int, optional
+        The font size to use for error labels
+
+    text_color : str, optional
+        The font color to use for error labels
+
+    text_ha : str, optional
+        The horizontal alignment modifier to use for error labels
+
+    Returns
+    -------
+    matplotlib.axes._axes.Axes
+    """
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError:
+        raise RuntimeError(_MATPLOTLIB_IMPORT_ERROR_MESSAGE)
+
+    # ambiguous error metric if both are provided
+    if error_bars is not None and conf_intervals is not None:
+        raise ValueError(_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR)
+    # TODO: Check if metrics and error_bars/conf_intervals same length 
+
+    if not (isinstance(metrics, list) or isinstance(metrics, str) or metrics is None):
+        raise ValueError(_METRICS_NOT_LIST_OR_STR_ERROR.format(type(metrics)))
+    
+    metrics = [metrics] if isinstance(metrics, str) else metrics
+    conf_intervals = [conf_intervals] if isinstance(conf_intervals, str) else conf_intervals
+    error_bars = [error_bars] if isinstance(error_bars, str) else error_bars
+
+    df = metric_frame.by_group
+
+    # only plot metrics that aren't tuples
+    if metrics is None:
+        metrics = []
+        for metric in list(df):
+            if not isinstance(df[metric][0], tuple) and metric != "_index":
+                metrics.append(metric)
+
+
+    if axs is None:
+        if subplot_shape is None:
+            subplot_shape = (1, len(metrics))
+        fig, axs = plt.subplots(*subplot_shape, squeeze=False, figsize=figsize)
+        axs = axs.flatten()
+
+    # plotting without error
```

Comment:
Nit: 'without error bars'. Might want to highlight this as a 'middle exit' too.

Reply:
What do you mean by "middle exit"?

Reply:
Usually returning from the middle of a subroutine (rather than from validation checks at the beginning, or the end) is regarded as a code smell. It's not _wrong_ but it can be hard to spot, and make it harder to understand what's going on. It's a case-by-case thing, since huge `if` blocks just to make everything go through a single return also have problems.